app/src/main/java/nl/hogeschoolrotterdam/projectb/fragment/MapFragment.java

```diff
@@ -149,18 +151,29 @@ public void onInfoWindowClick(Marker marker) {
             @Override
             public void onLocationResult(@Nullable Location location) {
                 if (location != null) {
-                    CameraPosition currentPosition = CameraPosition.builder().target(new LatLng(location.getLatitude(), location.getLongitude())).zoom(15).bearing(0).tilt(0).build();
+                    CameraPosition currentPosition = CameraPosition.builder().target(new LatLng(location.getLatitude(),
+                            location.getLongitude())).zoom(15).bearing(0).tilt(0).build();
                     map.moveCamera(CameraUpdateFactory.newCameraPosition(currentPosition));
                     map.setMyLocationEnabled(true);
                 }
             }
         });
         for (Memory memorie : Database.getInstance().getMemories()) {
-            Marker marker = googleMap.addMarker(new MarkerOptions().position(memorie.getLocation()).title(memorie.getTitle()).snippet((String) memorie.getDateText()));
+            Marker marker = googleMap.addMarker(new MarkerOptions().position(memorie.getLocation()).title(memorie.getTitle()).
+                    snippet((String) memorie.getDateText()).icon(bitmapDescriptorFromVector(getActivity(),memorie.getMemoryTypeIconId())));
             marker.setTag(memorie.getId());
         }
         inity();
     }
+    private BitmapDescriptor bitmapDescriptorFromVector(Context context, int vectorResId) {
+        Drawable vectorDrawable = ContextCompat.getDrawable(context, vectorResId);
+        vectorDrawable.setBounds(0, 0, vectorDrawable.getIntrinsicWidth(), vectorDrawable.getIntrinsicHeight());
+        Bitmap bitmap = Bitmap.createBitmap(vectorDrawable.getIntrinsicWidth(), vectorDrawable.getIntrinsicHeight(),
+                Bitmap.Config.ARGB_8888);
+        Canvas canvas = new Canvas(bitmap);
+        vectorDrawable.draw(canvas);
+        return BitmapDescriptorFactory.fromBitmap(bitmap);
+    }
```

Comment:
I think you should move this code to the `Memory` class with a function called `getTypeBitmap` or something similar. Then you can do `memory.getTypeBitmap(context)` on any page. 
It should not have a `vectorResId` parameter obviously, as it should use the res id already set in the memory.

Reply:
okidoki